### ISSUE TITLE
OOPless version

### ISSUE THREAD

**Issue by MyTDT-Mysoft** [buggins/dlangui#621]
there isnt a version of it for real programmers? not using stinky OOP?

**Comment by Superbelko** [comment]
Even though I also don't like the code and design, rewriting it even partially is basically meaningless, primarily because of lack of traction. 

On the other hand it is flexible and simple enough to understand, and there is no other UI library for D that allows one to quickly hack something customizable like dlangui. 
So if I need a GUI for my yet another crappy PoC tool that I will throw up after a week or two dlangui is my number 1 choice.

**Comment by MyTDT-Mysoft** [comment]
i see, yeah i wanted to really give a chance to D, because i liked many of things unlike C++ and the syntax isnt horrid like RUST and have nice ways to do what i use most (macros)... but unfortunately the fact that it even have OOP means that people will use and abuse of it in a non optional way... and the result is this... i can't find a GUI library without OOP :(

**Comment by Superbelko** [comment]
There is a reason for that.

Anyway you might want to look at immediate mode UI's like [ImGui](https://github.com/ocornut/imgui).
I have this bindings for that but it is not quite straightforward to set up, there is also alternative wrappers available in dub and on GitHub.

Unfortunately there is no declarative style alternatives like Flutter, at least not that I'm aware of.

https://github.com/Superbelko/imgui-d

**Comment by MyTDT-Mysoft** [comment]
> There is a reason for that.
> 
> Anyway you might want to look at immediate mode UI's like [ImGui](https://github.com/ocornut/imgui). I have this bindings for that but it is not quite straightforward to set up, there is also alternative wrappers available in dub and on GitHub.
> 
> Unfortunately there is no declarative style alternatives like Flutter, at least not that I'm aware of.
> 
> https://github.com/Superbelko/imgui-d

thanks... but wrappers against bloat-free bloated (because C++ / OOP) imgui wont do the trick...
i guess i will just use winapi directly in D, and add some D wrappers without OOP to simplify stuff and then implement winapi library for other platforms... since nobody cares about having a clean C usable gui library nowadays :) , i had to do for raspberyPi as well so no biggies... better than forcing OOP features...

at least D will bring me closer to also have native code (without having to handle Android Studio or related bloat), then possible i could code for android in both C and D

**Comment by GrimMaple** [comment]
`bindbc-imgui` gives you a pain-free experience in using imgui, with C-style API also!

There is no reason in rewriting `dlangui`, because it's currently in maintain-only mode. I'm going to keep up up-to-date with modern compilers and fix bugs, jsut slowly adding whatever I need

**Comment by MyTDT-Mysoft** [comment]
> `bindbc-imgui` gives you a pain-free experience in using imgui, with C-style API also!
> 
> There is no reason in rewriting `dlangui`, because it's currently in maintain-only mode. I'm going to keep up up-to-date with modern compilers and fix bugs, jsut slowly adding whatever I need

nope... even the samples of imgui is calling methods out of objects so its OOP CRAP like

ImFontConfig cfg;
cfg._default_ctor();

which is not tolerable to me.

**Comment by GrimMaple** [comment]
I'm closing this as won't fix :)
